Wikipedia/Code/SectionEditorViewController.swift

```diff
@@ -576,24 +576,24 @@ extension SectionEditorViewController: SectionEditorWebViewMessagingControllerSc
 
 extension SectionEditorViewController: SectionEditorInputViewsControllerDelegate {
     func sectionEditorInputViewsControllerDidTapMediaInsert(_ sectionEditorInputViewsController: SectionEditorInputViewsController) {
-        mediaWizardController = MediaWizardController(theme: theme, articleTitle: section?.titleText, siteURL: section?.article?.url.wmf_site)
-        mediaWizardController?.delegate = self
-        mediaWizardController?.prepare()
+        insertMediaViewController = InsertMediaViewController(articleTitle: section?.titleText, siteURL: section?.article?.url.wmf_site)
+        guard let insertMediaViewController = insertMediaViewController else {
+            return
+        }
+        insertMediaViewController.delegate = self
+        let navigationController = WMFThemeableNavigationController(rootViewController: insertMediaViewController, theme: theme)
+        present(navigationController, animated: true)
     }
 }
 
-extension SectionEditorViewController: MediaWizardControllerDelegate {
-    func mediaWizardController(_ mediaWizardController: MediaWizardController, didPrepareViewController viewController: UIViewController) {
-        present(viewController, animated: true)
-    }
-
-    func mediaWizardController(_ mediaWizardController: MediaWizardController, didTapCloseButton button: UIBarButtonItem) {
-        self.mediaWizardController = nil
+extension SectionEditorViewController: InsertMediaViewControllerDelegate {
+    func insertMediaViewController(_ insertMediaViewController: InsertMediaViewController, didTapCloseButton button: UIBarButtonItem) {
+        self.insertMediaViewController = nil
```

Comment:
You could get rid of this property now that you're using a view controller, the nav stack should keep it from deallocating.

Reply:
thanks, good catch